lib/karafka/core/monitoring/notifications.rb

```diff
@@ -117,27 +117,30 @@ def instrument(event_id, payload = EMPTY_HASH)
             return yield if assigned_listeners.empty?
```

Comment:
I'm thinking that we might also want to remove this early safeguard (`if assigned_listeners.empty?`) here above, given there is a similar safeguard in the `ensure` block below.
I understand that we are adding the overhead of two calls to `monotonic_now` in such case, in exchange for one less `assigned_listeners.empty?` check.